### ISSUE TITLE
System.Environment Is64BitProcess isn't available on MSBuildRuntimeType=Core

### ISSUE THREAD

**Issue by eerhardt** [dotnet/msbuild#7329]
### Issue Description
When running on `dotnet build` any MSBuild logic that uses `$([System.Environment]::Is64BitProcess` fails to load with error:

```
error MSB4185: The function "Is64BitProcess" on type "System.Environment" is not available for execution as an MSBuild property function.
```

### Steps to Reproduce
1. Create this project:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Is64 Condition="$([System.Environment]::Is64BitProcess)">true</Is64>
  </PropertyGroup>

</Project>
```
2. `dotnet build`

### Expected Behavior
It should build fine, and if you are using a 64-bit `dotnet.exe`, `Is64` property should be set to true.

### Actual Behavior
```
Test2.csproj(6,8): error MSB4185: The function "Is64BitProcess" on type "System.Environment" is not available for execution as an MSBuild property function.
```

### Analysis
It looks like this is only enabled on Desktop builds:

https://github.com/dotnet/msbuild/blob/518c041f4511a6bc23eb40703b69a94ea46c65fd/src/Build/Resources/Constants.cs#L308-L311
 
https://github.com/dotnet/msbuild/blob/73e25b6bd9eb235fe4dc1d2e623a9a613bbaa03b/src/Directory.BeforeCommon.targets#L19-L20

**Comment by akoeplinger** [comment]
I think this was because `Environment.Is64BitProcess` was only [added back in .NET Core 2.0](https://docs.microsoft.com/en-us/dotnet/api/system.environment.is64bitprocess?view=net-6.0#applies-to). Looks fine to me to expose now that msbuild targets later versions.

**Comment by akoeplinger** [comment]
A workaround might be to use `$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture`

**Comment by rainersigwald** [comment]
@akoeplinger Yup, that's it exactly. I'll do a pass over those not-in-.NET-Core-1.0 feature flags we have and see what we can turn back on.

**Comment by HavenDV** [comment]
This is also a problem when trying to build a WinUI 3 project with `dotnet build`:
`C:\Users\runneradmin\.nuget\packages\microsoft.windows.sdk.buildtools\10.0.22000.196\build\Microsoft.Windows.SDK.BuildTools.props(4,37): error MSB4185: The function "Is64BitProcess" on type "System.Environment" is not available for execution as an MSBuild property function.`

**Comment by rainersigwald** [comment]
@HavenDV the property function will be enabled in .NET SDK 6.0.300 (at some point in the future). However, since it's failing here, it's likely that other parts of `microsoft.windows.sdk.buildtools` will fail in `dotnet build`, so that may not get the whole scenario working by itself.

**Comment by eerhardt** [comment]
@rainersigwald - is it safe to assume that this fix will be released in the `6.0.300` SDK? Just trying to figure out if we need to work around it or not. If it is in `6.0.300`, then we shouldn't need to work around it.

**Comment by baronfel** [comment]
@eerhardt yeah, the new function is in the 6.0.300 previews available on dotnet/installer right now.

Sample:
```xml
<Project>
    <Target Name="Is64">
        <PropertyGroup>
            <Is64>$([System.Environment]::Is64BitProcess)</Is64>
        </PropertyGroup>
        <Message Text="Are we running x64? $(Is64)" />
    </Target>
</Project>
```

Result:

```posh
D:///msbuild-is64bitprocess [.NET] 6.0.300
16:03:54 ❯ dotnet --version; dotnet build -t:Is64 -v n
6.0.300-preview.22128.2
Microsoft (R) Build Engine version 17.2.0-preview-22123-01+047227bd7 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 3/1/2022 4:03:59 PM.
   1:2>Project "D:\code\Scratch\msbuild-is64bitprocess\64bit.proj" on node 1 (Is64 target(s)).
     1>Is64:
         Are we running x64? True
     1>Done Building Project "D:\code\Scratch\msbuild-is64bitprocess\64bit.proj" (Is64 target(s)).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.07
```
